### ISSUE TITLE
api to predict final zipfile size

### ISSUE THREAD

**Issue by thejoshwolfe** [thejoshwolfe/yazl#1]
for the purpose of providing streaming zip file downloads (for example), it would be useful to know how big the resulting zipfile would be when it's all done. the details of this api are not clear yet.

**Comment by sntran** [comment]
Sorry for bringing back an old issue, but I have a need for this and found this issue. Wasn't sure I should open a new issue.

Anyway, I see that we can get the final zipfile size in the callback to `.end`. However, I would like to know that size way before adding files, so I can pass it along the pipeline before creating the archive.

Would it be possible to add another API to return the predicted size for a list of inputs whose size is provided?

For example:

```js
yazl.size(
  fs.readdirSync("my/folder", { withFileTypes: true }).map(dirEntry => {
    return {
       ...dirEntry,
      size: fs.statSync(dirEntry.name).size,
    };
  }),
  {
    compress: false,
  },
);
```

Basically, it takes a list of file-like entries, with required `name` and `size` properties, and returns the final zipfile size.
